geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/tables/Z2Table.scala

```diff
@@ -141,43 +141,28 @@ object Z2Table extends GeoMesaTable {
       (0 until g.getNumPoints).map(g.getPointN).sliding(2).flatMap { case Seq(one, two) =>
         val (xmin, xmax) = minMax(one.getX, two.getX)
         val (ymin, ymax) = minMax(one.getY, two.getY)
-        zBox(xmin, ymin, xmax, ymax)
+        getZPrefixes(xmin, ymin, xmax, ymax)
       }.toSet
     case g: GeometryCollection => (0 until g.getNumGeometries).toSet.map(g.getGeometryN).flatMap(zBox)
     case g: Geometry =>
       val env = g.getEnvelopeInternal
-      zBox(env.getMinX, env.getMinY, env.getMaxX, env.getMaxY)
-  }
-
-  // gets a sequence of z values that cover the bounding box
-  private def zBox(xmin: Double, ymin: Double, xmax: Double, ymax: Double): Set[Long] = {
-    val zmin = Z2SFC.index(xmin, ymin).z
-    val zmax = Z2SFC.index(xmax, ymax).z
-    getZPrefixes(zmin, zmax)
+      getZPrefixes(env.getMinX, env.getMinY, env.getMaxX, env.getMaxY)
   }
 
   private def minMax(a: Double, b: Double): (Double, Double) = if (a < b) (a, b) else (b, a)
 
-  // gets z values that cover the interval
-  private def getZPrefixes(zmin: Long, zmax: Long): Set[Long] = {
-    val in = scala.collection.mutable.Queue((zmin, zmax))
-    val out = scala.collection.mutable.HashSet.empty[Long]
-
-    while (in.nonEmpty) {
-      val (min, max) = in.dequeue()
-      val ZPrefix(zprefix, zbits) = Z2.longestCommonPrefix(min, max)
-      if (zbits < GEOM_Z_NUM_BYTES * 8) {
-        // divide the range into two smaller ones using tropf litmax/bigmin
-        val (litmax, bigmin) = Z2.zdivide(Z2((min + max) / 2), Z2(min), Z2(max))
-        in.enqueue((min, litmax.z), (bigmin.z, max))
+  // gets z values that cover the bounding box
+  private def getZPrefixes(xmin: Double, ymin: Double, xmax: Double, ymax: Double): Set[Long] = {
+    Z2SFC.ranges((xmin, xmax), (ymin, ymax), 8 * GEOM_Z_NUM_BYTES).flatMap { range =>
+      val lower = range.lower & GEOM_Z_MASK
+      val upper = range.upper & GEOM_Z_MASK
+      if (lower == upper) {
```

Comment:
ok, it shouldn't be possible due to the `GEOM_Z_MASK`. They will either be equal, or be at least `GEOM_Z_STEP` apart.